param/parameterized.py

```diff
@@ -1600,66 +1612,193 @@ def __setstate__(self,state):
             setattr(self,name,value)
         self.initialized=True
 
+    def __repr__(self):
+        """
+        Provide a nearly valid Python representation that could be used to recreate
+        the item with its parameters, if executed in the appropriate environment.
 
-    @classmethod
-    def params(cls,parameter_name=None):
+        Returns 'classname(parameter1=x,parameter2=y,...)', listing
+        all the parameters of this object.
```

Comment:
What's the difference between repr and script_repr?  I remember there was one at some time, but can no longer remember why we needed script_repr to be distinct.